src/shared/libfido2-util.c

```diff
@@ -256,6 +256,11 @@ static int fido2_use_hmac_hash_specific_token(
                                        "PIN required to unlock, but FIDO2 device %s does not support it.",
                                        path);
 
+        if (!has_up && up_required)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
```

Comment:
maybe for both these cases use EHWPOISON, to indicate the hw doesn't meet our expectations